lib/ansible/modules/files/file.py

```diff
@@ -196,6 +196,16 @@
     owner: foo
     group: foo
 
+- name: Remove file (delete file)
+  file:
+    path: /etc/foo
+    state: absent
+
+- name: Recursively remove directory
```

Comment:
I don't feel this is correct. This looks like the same code twice. This make need `recurse: yes` in order to delete directories.

Reply:
I was mistaken, it looks like recurse can only be used with state 'directory'.

Since these are the same example though, it could probably just be combined.

Reply:
I tried these and they worked as I pasted them here.

Reply:
I just had some time to look at this more closely and this is working correctly. As @s-hertel said, `recure: yes` is only valid for `state: directory` but isn't needed when removing directories.